### ISSUE TITLE
add epochs training

### ISSUE THREAD

**Issue by seungbo** [ultralytics/ultralytics#16402]
### Search before asking

- [x] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/ultralytics/ultralytics/discussions) and found no similar questions.

### Question

I did 4 epochs with yolov8. I tried an additional 10 epochs by applying “resume=True”, but I got this error: “..\yolo\runs\detect\train11\weights\best.pt training to 4 epochs is finished, nothing to resume.” I would like to know how to train additional epochs on a model trained with 4 epochs.

**Comment by UltralyticsAssistant** [comment]
👋 Hello @seungbo, thank you for reaching out to Ultralytics 🚀! 

It seems you're trying to continue training with additional epochs. For this, please ensure you have the correct settings and paths specified. If this is a 🐛 Bug Report, providing a [minimum reproducible example](https://docs.ultralytics.com/help/minimum_reproducible_example/) would be greatly helpful for us to understand the issue better.

Meanwhile, you can check our comprehensive [Docs](https://docs.ultralytics.com) for guidance or explore the [CLI](https://docs.ultralytics.com/usage/cli/) and [Python](https://docs.ultralytics.com/usage/python/) tutorials to ensure all steps are correctly followed.

An Ultralytics engineer will review your issue soon, but feel free to join our community for further discussions or support:
- [Discord](https://ultralytics.com/discord) 🎧
- [Discourse](https://community.ultralytics.com)
- [Subreddit](https://reddit.com/r/ultralytics)

## Upgrade

Make sure your `ultralytics` package is the latest version with all dependencies:

```bash
pip install -U ultralytics
```

Remember to verify your environment by reviewing the [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml) and ensure you are using a [**Python>=3.8**](https://www.python.org/) environment with [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/).

## Environments

YOLOv8 can be run smoothly in these environments:

- **Notebooks**: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/ultralytics/yolov8"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud**: [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon**: [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker**: [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml/badge.svg" alt="Ultralytics CI"></a>

If green, our [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml?query=event%3Aschedule) tests are currently passing, ensuring all YOLOv8 [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) are functioning well across various systems.

We appreciate your patience and collaboration 💡!

**Comment by Y-T-G** [comment]
It's not possible to "add epochs" like that. You can only set the number of epochs when you start the training. `resume` only resumes training if it got interrupted before it finished the epoch you first used.

**Comment by seungbo** [comment]
@Y-T-G So is there a way to add more epochs to train more?

**Comment by Y-T-G** [comment]
You can reuse the model by passing the location to your `best.pt` in `model=best.pt` and then train more epochs. You also need to modify [some parameters](https://redd.it/1eolwl8).

**Comment by seungbo** [comment]
@Y-T-G `model = YOLO("../yolo/runs/detect/train11/weights/best.pt")
results = model.train(data="data.yaml", epochs=4, imgsz=640, workers=0, verbose=True, batch=32, plots=True, warmup_epochs=0)`
I did this, but how do I set 'lr0'?

**Comment by Y-T-G** [comment]
You can pass it as argument to `model.train()`. You also have to pass an `optimizer` such as `"SGD"` or `"AdamW"`
